src/main/java/net/sllmdilab/t5/converters/PCD_01MessageToSqlObservationsConverter.java

```diff
@@ -0,0 +1,182 @@
+			Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+			PCD01Message message = (PCD01Message) unmarshaller.unmarshal(document);
+
+			final List<SqlObservation> result = new ArrayList<>();
+
+			message.getPatientResult().forEach(patientResult -> {
+				patientResult.getOrderObservations().forEach(orderObs -> {
+					orderObs.getMDS().forEach(mds -> {
+						result.addAll(handleLevel("MDS", mds.getVMDAndObservation(), new LinkedList<>()));
+					});
+				});
+			});
+
+			return result;
+		} catch (JAXBException e) {
+			throw new T5Exception("Message parsing failed.", e);
+		}
+	}
+
+	private List<SqlObservation> handleLevel(String level, List<Object> objects, Deque<IdItem> idStack) {
+		List<SqlObservation> result = new LinkedList<>();
+
+		List<IdItem> items = findIds(level, objects);
+		items.forEach(item -> idStack.push(item));
+
+		result.addAll(getSqlObservations(level, objects, idStack));
+
+		objects.forEach(object -> {
+			if (object instanceof VMD) {
+				result.addAll(handleLevel("VMD", ((VMD) object).getCHANAndObservation(), idStack));
+			} else if (object instanceof CHAN) {
+				result.addAll(handleLevel("CHAN", ((CHAN) object).getMetricAndObservation(), idStack));
+			} else if (object instanceof Metric) {
+				result.addAll(handleLevel("Metric", ((Metric) object).getFacetAndObservation(), idStack));
+			} else if (object instanceof Facet) {
+				result.addAll(handleLevel("Facet", Arrays.asList(((Facet) object).getObservation()), idStack));
+			}
+		});
+
+		for (int i = 0; i < items.size(); ++i) {
+			idStack.pop();
+		}
+
+		return result;
+	}
+
+	private List<SqlObservation> getSqlObservations(String level, List<Object> objects, Deque<IdItem> idStack) {
+		List<SqlObservation> result = new LinkedList<>();
+		objects.forEach(object -> {
+			if (object instanceof Observation) {
+				Observation obs = (Observation) object;
+				SqlObservation sqlObservation = new SqlObservation();
+
+				if (obs.getValue() != null && !obs.getValue().isEmpty()) {
+					sqlObservation.setValue(obs.getValue().get(0).getValue());
+					sqlObservation.setValueType(obs.getValue().get(0).getTypeHL7V2());
+				}
+
+				for (ObsIdentifier identifier : obs.getObsIdentifier()) {
+					if (identifier.isIsAlternate() != Boolean.TRUE) {
+						sqlObservation.setCode(identifier.getValue());
+						sqlObservation.setCodeSystem(identifier.getCodingSystemName());
+						break;
+					}
+				}
+
+				if (obs.getTimestamp() != null) {
+					sqlObservation.setStartTime(obs.getTimestamp().getValue().toGregorianCalendar(TimeZone.getTimeZone("UTC"), null, null).getTime());
+				}
+
+				if (obs.getUnit() != null) {
+					String value = obs.getUnit().getValue();
+					if (!StringUtils.isBlank(value)) {
+						sqlObservation.setUnit(value);
+					}
+
+					String system = obs.getUnit().getCodingSystemName();
+					if (!StringUtils.isBlank(system)) {
+						sqlObservation.setUnitSystem(system);
+					}
+				}
+
```

Comment:
Some of these statements could be separated from getSqlObservations (or commented) to make all steps more clear.